30_os-prober: add the common commands to the generated entries

This is a simplified version of grub-2.00.30_os-prober.options.patch by
Alex Burmashev <alex.burmashev@gmail.com>.

The patch adds a few common commands (like "load_video") to the entries
generated by 30_os-prober. These commands are usually present in the entries
generated by 10_linux, so let us make things uniform here.

I have removed the hunk for efifb and such - it was almost never used anyway
and may be buggy, who knows.

Signed-off-by: Evgenii Shatokhin <eshatokhin@virtuozzo.com>

diff --git a/util/grub.d/30_os-prober.in b/util/grub.d/30_os-prober.in
index d577046f0..0263b5811 100644
--- a/util/grub.d/30_os-prober.in
+++ b/util/grub.d/30_os-prober.in
@@ -41,6 +41,14 @@ if [ -z "${OSPROBED}" ] ; then
   exit 0
 fi
 
+add_options() {
+  if [ "x$GRUB_GFXPAYLOAD_LINUX" != xtext ]; then
+    echo "        load_video" | sed "s/^/$submenu_indentation/"
+  fi
+  echo "        set gfxpayload=$GRUB_GFXPAYLOAD_LINUX" | sed "s/^/$submenu_indentation/"
+  echo "        insmod gzio" | sed "s/^/$submenu_indentation/"
+}
+
 osx_entry() {
     if [ x$2 = x32 ]; then
         # TRANSLATORS: it refers to kernel architecture (32-bit)
@@ -310,6 +319,7 @@ $(save_default_entry | sed -e "s/^/\t/"; add_options; printf '%s\n' "${prepare_b
 menuentry '$(echo "$title $onstr" | grub_quote)' $CLASS --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-simple-$boot_device_id' {
 EOF
 	    save_default_entry | grub_add_tab
+	    add_options
 	    printf '%s\n' "${prepare_boot_cache}"
 	    cat <<  EOF
 	echo '$message_kernel'
@@ -344,6 +354,7 @@ EOF
 	menuentry '$(echo "$title $onstr" | grub_quote)' --class gnu-linux --class gnu --class os \$menuentry_id_option 'osprober-gnulinux-$LKERNEL-${recovery_params}-$boot_device_id' {
 EOF
 	save_default_entry | sed -e "s/^/$grub_tab$grub_tab/"
+	add_options
 	printf '%s\n' "${prepare_boot_cache}" | grub_add_tab
 	cat <<  EOF
 		echo '$message_kernel'
